src/systempolicy/systemPolicy.go

```diff
@@ -844,6 +844,10 @@ func ConvertWPFSToKnoxSysPolicy(wpfsSet types.ResourceSetMap, pnMap types.Policy
 
 func getPodInstance(key SysLogKey, pods []types.Pod) (types.Pod, error) {
 	for _, pod := range pods {
+		if key.Namespace != "" || key.PodName != "" {
+			return pod, nil
+		}
```

Comment:
I have doubts about this check. A non-nil value does not mean that the pod matches it. Didn't understand the rational for this check. This check might impact existing behaviour in case of k8s orchestrated workloads.